esphome/components/max31865/max31865.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include "esphome/core/component.h"
+#include "esphome/components/sensor/sensor.h"
+#include "esphome/components/spi/spi.h"
+
+namespace esphome {
+namespace max31865 {
+
+class MAX31865Sensor : public sensor::Sensor,
+                       public PollingComponent,
+                       public spi::SPIDevice<spi::BIT_ORDER_MSB_FIRST, spi::CLOCK_POLARITY_LOW,
+                                             spi::CLOCK_PHASE_TRAILING, spi::DATA_RATE_4MHZ> {
+ public:
+  void setup() override;
+  void dump_config() override;
+  float get_setup_priority() const override;
+  void set_pin_count(bool even = false) { this->even_pins_ = even; }
```

Comment:
```suggestion
  void set_pin_count(bool even) { this->even_pins_ = even; }
```

Having an optional here makes no sense - setters accept a value.